### ISSUE TITLE
french dictionary

### ISSUE THREAD

**Issue by damoleon** [monich/harbour-wordle#11]
Hello,

I'm attaching two files hoping that they could be use to add french dictionary.
Thanks in advance
[words-fr.zip](https://github.com/monich/harbour-wordle/files/8056434/words-fr.zip)
e

**Comment by monich** [comment]
I'm not seeing any words with accented vowels, e.g. **passé**. That's suspicious.

I would prefer the list to be utf-8 encoded and lowercased (even though the letters are displayed in the upper case, app internally operates with lower case letters). It's perfectly fine to use accented and all kinds of weird language specific characters.

wordle-tool will remove all the spaces, newlines etc. and convert the source text files to the target encoding (presumably windows-1252) when it generates `words` and `xwords` files that will be actually loaded by the app.

**Comment by monich** [comment]
![fr_001](https://user-images.githubusercontent.com/5909522/153794327-d764a817-9909-4900-b1da-ca24201f4cff.png)

And what do you think about this keypad layout for French?

**Comment by damoleon** [comment]
Yes the keyboard is fine, although the "enter" key at the left side is not intuitive imho.
Regarding accentuation , it's generally not used in french for this kind of games.
See https://www.youtube.com/channel/UCDfg9spaooHe29Lk_imC9LQ
https://sutom.nocle.fr/#
"passe" and "passé" are two different word. Both would be accepted.
![Capture d'écran_20220214_001](https://user-images.githubusercontent.com/6343225/153830880-f94a9e4c-1b35-4119-b0e3-9aacbf6a2a00.png)

**Comment by monich** [comment]
So, if you think there's no need for accented characters. Then I'll remove those from the keypad.

I just pushed a WIP branch https://github.com/monich/harbour-wordle/issues/5 so that you get an idea how it's going to look like.

**Comment by baxyz** [comment]
> Hello,
> 
> I'm attaching two files hoping that they could be use to add french dictionary. Thanks in advance [words-fr.zip](https://github.com/monich/harbour-wordle/files/8056434/words-fr.zip) e

I think `words` and `xwords` are inverted... or maybe I miss something.

**Comment by damoleon** [comment]
"To make a dictionary, you basically need two lists of 5-letter words. One (called words) contains the words to be guessed, and the second one (xwords) is the list of the additional words that can be used for checking." from openrepos
So I don't think they are inverted. The list can be improved though, for sure. It was a draft to see if it was possible to add french.
I'll be happy to discuss, improve the list of words to be guessed :) There was a typo with "zombi" in the words one. I upload a new one here with all words lowercase
[words2.txt](https://github.com/monich/harbour-wordle/files/8074107/words2.txt)
.

**Comment by monich** [comment]
OK, I pushed that to master.

**Comment by monich** [comment]
And released as [1.0.5](https://github.com/monich/harbour-wordle/releases/tag/1.0.5)!

**Comment by damoleon** [comment]
Great ! Thank you very much :) !
Is there a way to easily improve the list of words ?

**Comment by monich** [comment]
Here is the workflow which I had in mind

1. Fork this app
2. Create a work branch
3. Modify the source text files [here](https://github.com/monich/harbour-wordle/tree/master/dict/fr)
4. Run make in that directory. That updates the actual runtime files [here](https://github.com/monich/harbour-wordle/tree/master/data/fr)
5. Commit the modified files and submit a PR
6. Once the PR is merged, delete the work branch, fast forward your master and you're back at step 2

Or you can just modify the source .txt files and submit just that. I can do the rest.

**Comment by damoleon** [comment]
sorry you lost me ;) I'd go for the last option "Or you can just modify the source .txt files and submit just that. I can do the rest." :)
Now how do I modify and then submit the source .txt file ? here : 
![fork](https://user-images.githubusercontent.com/6343225/154488508-84d08a86-1a93-443c-ba1c-16d63943a2a4.png)
Or I simply upload here as a comment the updated .txt file ? (I won't update it every day ;)
I just don't want to break anything of your work ;)
Sorry I'm not a dev and not familiar with github processes :/

**Comment by monich** [comment]
I mean something like this:

1. (make a fork via github web interface)
2. git clone git@github.com:damoleon/harbour-wordle.git
3. cd harbour-wordle
4. git checkout -b wip
5. vi dict/fr/words.txt
6. git commit dict/fr/words.txt
7. git push -u origin wip
8. (submit a PR via github web interface)

But if you're not comfortable with git, you can download the current words.txt (or xwords.txt), modify them, create an issue and attach the files. That would work too, but in this case I'd have to do the git magic myself. Which is not a big deal.
